GCMessengerSDKSample/GCMessengerSDKSample/AccountDetailsViewController.swift

```diff
@@ -38,26 +38,44 @@ class AccountDetailsViewController: UIViewController {
         
         loggingSwitch.setOn(UserDefaults.logging, animated: true)
     }
-
+    
     @IBAction func startChatButtonTapped(_ sender: UIButton) {
+        validateFields(shouldStartChat: true)
+    }
+    
+    @IBAction func chatAvailabilityButtonTapped(_ sender: UIButton) {
+        validateFields(shouldStartChat: false)
+    }
+    
+    private func validateFields(shouldStartChat: Bool) {
```

Comment:
I suggest making this function to have a single responsibility of validating form fields, rather than being responsible for other tasks such as starting a chat or checking chat availability. By separating concerns in this way, the code becomes more modular, easier to test and maintain.